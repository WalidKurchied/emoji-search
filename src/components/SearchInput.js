import React, { Component } from 'react'

class SearchInput extends Component {
    handleFilterOnChange = e => {
        this.props.filterEmojis(e.target.value)
    }

    render() {
        return (
            <div id="searchInputContainer">
                <input id="searchInput" type="text" onChange={this.handleFilterOnChange} />
            </div>
        )
    }
}

export default SearchInput;
