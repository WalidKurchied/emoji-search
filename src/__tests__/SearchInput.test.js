import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import SearchInput from '../components/SearchInput';

import '../setupTests';

it('Should match snapshot', () => {
    const component = create(<SearchInput />);
    expect(component.toJSON()).toMatchSnapshot();
});

it('Should render input element', () => {
    const component = shallow(<SearchInput />);
    expect(component.find('input')).toHaveLength(1);
});

it('Should perform on input change event successfully', () => {
    const mockedFilterEmojis = jest.fn();
    
    const component = shallow(<SearchInput filterEmojis={mockedFilterEmojis} />);

    component.find('input').simulate('change', {target: 'earth'});

    expect(mockedFilterEmojis.mock.calls.length).toBe(1);
});