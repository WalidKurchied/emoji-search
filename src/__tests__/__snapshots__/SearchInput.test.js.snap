// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<div
  id="searchInputContainer"
>
  <input
    id="searchInput"
    onChange={[Function]}
    type="text"
  />
</div>
`;
